tests/integrate_test/iiss/decentralized/test_preps_divide_node_address.py

```diff
@@ -0,0 +1,243 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018 ICON Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""IconScoreEngine testcase
+"""
+from typing import TYPE_CHECKING
+
+from iconservice import Address
+from iconservice.base.exception import ExceptionCode
+from iconservice.icon_constant import ICX_IN_LOOP, PREP_MAIN_PREPS, ConfigKey, Revision
+from tests import create_address
+from tests.integrate_test.iiss.test_iiss_base import TestIISSBase
+from tests.integrate_test.test_integrate_base import EOAAccount
+
+if TYPE_CHECKING:
+    pass
+
+
+class TestPreps(TestIISSBase):
+    def _make_init_config(self) -> dict:
+        config: dict = super()._make_init_config()
+        config[ConfigKey.PREP_REGISTRATION_FEE] = 0
+        return config
+
+    def setUp(self):
+        super().setUp()
+        self.init_decentralized()
+
+    def test_prep_register_node_address_before_rev_DIVIDE_NODE_ADDRESS(self):
+        account: 'EOAAccount' = self.create_eoa_account()
+        self.distribute_icx(accounts=[account],
+                            init_balance=1 * ICX_IN_LOOP)
+
+        dummy_node: 'Address' = create_address()
+
+        # register prep
+        reg_data: dict = self.create_register_prep_params(account)
+        reg_data["nodeAddress"] = str(dummy_node)
+
+        tx: dict = self.create_register_prep_tx(from_=account,
+                                                reg_data=reg_data)
+
+        _, tx_results, _, _, main_prep_as_dict = self.debug_make_and_req_block(tx_list=[tx])
+        self.assertEqual(tx_results[1].failure.code, ExceptionCode.INVALID_PARAMETER)
+        self.assertEqual(tx_results[1].failure.message, f"Mismatch nodeAddress and address. "
+                                                        f"'divide node address' revision need to be accepted"
+                                                        f"(revision: 9)")
```

Comment:
In my opinion, to check error message is too specific.

Reply:
Our exception type is too general. like(invalid parameter)
so I think it is reasonable case